test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -12627,13 +12647,17 @@ def __init__(self, name=None, attrs={}, pbsconf_file=None, diagmap={},
                                     'config.d')
         self.config = {}
         if self.platform == 'cray' or self.platform == 'craysim':
+            usecp = os.path.realpath('/home')
             if self.platform == 'cray':
                 alps_client = self.du.which(exe='apbasil')
+                if os.path.exists('/opt/cray/alps/default/bin/apbasil'):
+                    alps_client = '/opt/cray/alps/default/bin/apbasil'
```

Comment:
Thanks @borlesanket, one more request.  Rather than do the "which" command all the time at line 12652, it would be better if you made an else case for os.path.exists and put the which apbasil in there.